File: scrumly-backend/scrumly-api/event-service/src/main/java/com/scrumly/eventservice/services/EventService.java
package com.scrumly.eventservice.services;

import com.scrumly.eventservice.domain.events.EventEntity;
import com.scrumly.eventservice.dto.events.EventDto;
import com.scrumly.eventservice.dto.requests.events.CreateActivityRQ;
import com.scrumly.specification.PageDto;
import com.scrumly.specification.SearchQuery;

import java.util.List;

public interface EventService {
    List<EventDto> getAllEvents();

    List<EventDto> getAllEvents(String createdFor);

    PageDto<EventDto> findEvents(SearchQuery searchQuery);

    EventDto getEventById(String eventId);

    List<EventDto> createEvent(CreateActivityRQ createActivityRQ);

    List<EventEntity> scheduleEvent(CreateActivityRQ createActivityRQ);

    EventDto updateEvent(String eventId, EventDto eventDTO);

    EventDto scheduleCalendarEvent(String eventId, CreateActivityRQ rq);

    void cancelEvent(String eventId);

    void cancelEvents(List<String> eventId);

    void cancelRecurrentEvent(String eventId, String recurrenceEventId);

    void cancelAllRecurrentEvent(String recurrenceEventId);

    void deleteEvent(String eventId);

    void deleteEvents(List<String> events);
}

File: scrumly-ui/src/app/member-dashboard/modules/events/services/event.service.ts
import { Injectable } from '@angular/core';
import { environment } from "../../../../../enviroments/enviroment";
import { HttpClient } from "@angular/common/http";
import { BehaviorSubject, Observable } from "rxjs";
import { PageDto, SearchQuery } from "../../../../ui-components/models/search-filter.model";
import { CreateActivityRQ, EventDto } from "../model/event.model";
import { ActivityDto } from "../model/activity.model";
import { EventPreSelection } from "../components/schedule-event-sidebar/schedule-event-sidebar.component";

@Injectable({
  providedIn: 'root'
})
export class EventService {
  scheduleEventSidebarState: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
  onScheduleEventSidebarState = this.scheduleEventSidebarState.asObservable();

  templateSelectionState: BehaviorSubject<EventPreSelection| undefined> = new BehaviorSubject<EventPreSelection | undefined>(undefined);
  onTemplateSelectionState = this.templateSelectionState.asObservable();

  editEventSidebarState: BehaviorSubject<ActivityDto | undefined> = new BehaviorSubject<ActivityDto | undefined>(undefined);
  onEditEventSidebarState = this.editEventSidebarState.asObservable();

  private apiUrl = `${ environment.api_url }/events/event`;

  constructor(private http: HttpClient) {
  }

  getAllEvents(): Observable<EventDto[]> {
    return this.http.get<EventDto[]>(this.apiUrl);
  }

  getAllEventsByCreatedFor(createdFor: string): Observable<EventDto[]> {
    return this.http.get<EventDto[]>(`${ this.apiUrl }/createdFor/${ createdFor }`);
  }

  findEvents(searchQuery: SearchQuery): Observable<PageDto<EventDto>> {
    return this.http.post<PageDto<EventDto>>(`${ this.apiUrl }/search`, searchQuery);
  }

  getEventById(eventId: string): Observable<EventDto> {
    return this.http.get<EventDto>(`${ this.apiUrl }/${ eventId }`);
  }

  createEvent(createEventRQ: CreateActivityRQ): Observable<EventDto> {
    return this.http.post<EventDto>(this.apiUrl, createEventRQ);
  }

  updateEvent(eventId: string, eventDTO: EventDto): Observable<EventDto> {
    return this.http.put<EventDto>(`${ this.apiUrl }/${ eventId }`, eventDTO);
  }

  deleteEvent(eventId: string): Observable<void> {
    return this.http.delete<void>(`${ this.apiUrl }/${ eventId }`);
  }
}
